app/src/main/java/fr/neamar/kiss/result/AppResult.java

```diff
@@ -268,6 +281,99 @@ public void run() {
         dialog.show();
     }
 
+    private void launchRenameDialog(final Context context, RecordAdapter parent, final AppPojo app) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(context.getResources().getString(R.string.app_rename_title));
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            builder.setView(R.layout.rename_dialog);
+        } else {
+            builder.setView(View.inflate(context, R.layout.rename_dialog, null));
+        }
+
+        builder.setPositiveButton(R.string.custom_name_rename, (dialog, which) -> {
+            EditText input = ((AlertDialog) dialog).findViewById(R.id.rename);
+            dialog.dismiss();
+
+            // Set new name
+            String newName = input.getText().toString().trim();
+            app.setName(newName);
+            KissApplication.getApplication(context).getDataHandler().renameApp(app.getComponentName(), newName);
+
+            // Show toast message
+            String msg = context.getResources().getString(R.string.app_rename_confirmation, app.getName());
+            Toast.makeText(context, msg, Toast.LENGTH_SHORT).show();
+
+            // We'll need to reset the list view to its previous transcript mode,
+            // but it has to happen *after* the keyboard is hidden, otherwise scroll will be reset
+            // Let's wait for half a second, that's ugly but we don't have any other option :(
+            final Handler handler = new Handler();
+            handler.postDelayed(() -> parent.updateTranscriptMode(AbsListView.TRANSCRIPT_MODE_ALWAYS_SCROLL), 500);
+        });
+        builder.setNeutralButton(R.string.custom_name_set_default, (dialog, which) -> {
```

Comment:
Shouldn't this action be the setNegativeButton, and the "cancel" be setNeutralButton instead?